src/analysis/special_functions/trigonometric.lean

```diff
@@ -2205,6 +2250,26 @@ begin
     ring },
 end
 
+lemma tan_add {x y : ℂ} (hx : ∀ k : ℤ, x ≠ (2 * k + 1) * π / 2)
+  (hy : ∀ k : ℤ, y ≠ (2 * k + 1) * π / 2) :
```

Comment:
This one is a bit strange...the theorem should hold even if `cos x` and `cos y` both equal `0` but I can't figure out a way to prove it that does not require those facts. The only time the theorem breaks is when one of `cos x` and `cos y` is equal to `0` but the other is not. 

The inclusion of assumptions `hx` and `hy` here necessitates not-so-user-friendly assumptions in `tan_eq` and `tan_eq'` below.

Reply:
I don't think I was sufficiently clear: I can modify this lemma by replacing `hx` and `hy` with a single assumption `(h : ((∀ k : ℤ, x ≠ (2 * k + 1) * π / 2) ∧ ∀ l : ℤ, y ≠ (2 * l + 1) * π/ 2) ∨ ((∃ k : ℤ, x = (2 * k + 1) * π / 2) ∧ ∃ l : ℤ, y = (2 * l + 1) * π / 2))` and still be able to prove it. Although this would definitely be a more precise formulation of the theorem, I'm not sure it is more useful; in fact, I think one could argue that it makes the lemma somewhat unwieldily. I don't have a personal preference.

I have already written the version with the single assumption so it would not be much trouble for me to switch it in for this one if it is deemed preferable.

Reply:
I think that `hx : cos x ≠ 0` is more readable than `(hx : ∀ k : ℤ, x ≠ (2 * k + 1) * π / 2)`

Reply:
I agree, but we've had [that conversation](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.233746.20deriv.20tan/near/208405818) before.

Reply:
I've changed my mind, I think that the new version of `tan_add` that I described above is preferable due to the improved accuracy of its statement. Additionally, I've tried using that version in writing some other lemmas and did not find it as unwieldily as I predicted. I'll make push this update soon.